### PR TITLE
Refactor `==` operator for Collection of Equatable

### PR DESCRIPTION
**Pull request by ikesyo**

### CODE REVIEW

File: Sources/Delta.swift [hunk 1]
```diff
@@ -31,12 +31,15 @@ extension Delta where Snapshot.Iterator.Element: Equatable, ChangeRepresentation
 fileprivate extension Collection where Iterator.Element: Equatable {
 
 	fileprivate static func ==(lhs: Self, rhs: Self) -> Bool {
-
-		guard lhs.count == rhs.count && lhs.count == rhs.count
```
Comment (ikesyo) [inline Sources/Delta.swift:35]: > `&& lhs.count == rhs.count`

Removed duplicated count check.

File: Sources/Delta.swift [hunk 2]
```diff
@@ -31,12 +31,15 @@ extension Delta where Snapshot.Iterator.Element: Equatable, ChangeRepresentation
...
-			else { return false }
-
-		guard zip(lhs, rhs).map(==).reduce(true, { $0 && $1 })
-			else { return false }
+		guard lhs.count == rhs.count else {
+			return false
+		}
+
+		for (left, rigtht) in zip(lhs, rhs) {
+			if left != right {
```
Comment (ikesyo) [inline Sources/Delta.swift]: Short-circuiting.

File: Sources/Delta.swift (hunk 1 above)
Comment (dmcrodrigues) [reply Sources/Delta.swift:35]: Good catch!

File: Sources/Delta.swift (hunk 2 above)
Comment (dmcrodrigues) [reply Sources/Delta.swift]: 👍 but maybe we could use `first(where:)`?

```swift
return zip(lhs, rhs).first(where: !=) == nil
```

**Review by dmcrodrigues — APPROVED** [review]

**Comment by dmcrodrigues** [comment]
@ikesyo sorry but can you remove your last commit (898bc92)? I would to fix that in a separate PR because there are a few more changes to do.

**Comment by ikesyo** [comment]
Just rebased.

**Comment by dmcrodrigues** [comment]
Awesome thanks @ikesyo 🎉 ✨

**Pull request merged**